src/fnft_nsep.c

```diff
@@ -198,15 +231,30 @@ static inline INT gridsearch(const UINT D,
     INT W = 0, *W_ptr = NULL;
     UINT K, K_filtered;
     UINT M;
-    UINT i;
+    UINT i, upsampling_factor, D_effective;// upsampling_factor*D gives the effective number of samples
     INT ret_code = SUCCESS;
-
+    COMPLEX *q_preprocessed = NULL;
+    COMPLEX *r_preprocessed = NULL;
+    UINT Dsub = 0;
+    UINT first_last_index[2] = {0};
     // Check inputs
     if (sheet_indices != NULL)
         return E_NOT_YET_IMPLEMENTED(sheet_indices, "Pass NULL");
+    
+    upsampling_factor = nse_discretization_upsampling_factor(opts_ptr->discretization);
```

Comment:
Error check missing (signaled by `upsampling_factor == 0'), maybe also in other places.

Reply:
Good catch!

Reply:
Leaving this conversation open because it is possible I may have missed it in some place.